charm/src/iterate.ts

```diff
@@ -90,13 +90,17 @@ export const generateNewRecipeVersion = (
     name,
   });
 
-  return compileAndRunRecipe(
+  const newCharm = await compileAndRunRecipe(
     charmManager,
     newRecipeSrc,
     newSpec,
-    charm.getSourceCell()?.key("argument"),
+    parent.getSourceCell()?.key("argument"),
     recipeId ? [recipeId] : undefined,
   );
+
+  newCharm.getSourceCell(charmSourceCellSchema)?.key("parents").push(parent);
```

Comment:
I wonder if we could have passed parent through to the generation of the charm and have it happen there ... but good for now